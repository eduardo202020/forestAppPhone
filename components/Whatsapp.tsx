import React from 'react';
import { View, Alert, Text, StyleSheet } from 'react-native';
import * as Linking from 'expo-linking';
import { IconButton } from 'react-native-paper';

const openWhatsApp = () => {
  const phoneNumber = '+51991004126'; // Reemplaza con el número de teléfono
  const message = 'Hola, tengo una consulta.'; // Reemplaza con el mensaje predeterminado

  const url = `whatsapp://send?phone=${phoneNumber}&text=${encodeURIComponent(message)}`;

  Linking.openURL(url).catch(() => {
    Alert.alert('Error', 'No se pudo abrir WhatsApp.');
  });
};

export function WhatsAppButton() {
  return (
    <View style={styles.container}>
      <IconButton
        onPress={openWhatsApp}
        icon="whatsapp"
        size={35}
        iconColor="#25D366"
      />
      <View>
        <Text style={styles.text}>Contactar por WhatsApp</Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontSize: 30,
    color: '#25D366',
  },
});
