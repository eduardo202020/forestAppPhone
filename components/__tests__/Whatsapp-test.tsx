import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import { WhatsAppButton } from '../Whatsapp';

function Whatsapp() {
  return <Text>Test component funcionando</Text>;
}

// Mock de react-native-gesture-handler
test('TestComponent se renderiza sin errores', () => {
  const tree = renderer.create(<Whatsapp />).toJSON();
  expect(tree).toMatchSnapshot();
});

jest.mock('react-native-paper', () => {
  const Real = jest.requireActual('react-native-paper');
  return {
    ...Real,
    IconButton: () => <div>icon</div>,
  };
});
jest.mock('@expo/vector-icons', () => ({ FontAwesome: () => <div>icon</div> }));

describe('WhatsAppButton', () => {
  it('renderiza correctamente el botÃ³n y el texto', () => {
    const tree = renderer.create(<WhatsAppButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
