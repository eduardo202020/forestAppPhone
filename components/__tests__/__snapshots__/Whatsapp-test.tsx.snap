// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent se renderiza sin errores 1`] = `
<Text>
  Test component funcionando
</Text>
`;

exports[`WhatsAppButton renderiza correctamente el bot√≥n y el texto 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <div>
    icon
  </div>
  <View>
    <Text
      style={
        {
          "color": "#25D366",
          "fontSize": 30,
        }
      }
    >
      Contactar por WhatsApp
    </Text>
  </View>
</View>
`;
